x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -213,86 +206,11 @@ public Factory(Client client, ClusterService clusterService, Settings settings)
         @Override
         public void accept(ClusterState state) {
             minNodeVersion = state.nodes().getMinNodeVersion();
-            currentInferenceProcessors = countNumberInferenceProcessors(state);
-        }
-
-        public static int countNumberInferenceProcessors(ClusterState state) {
-            Metadata metadata = state.getMetadata();
-            if (metadata == null) {
-                return 0;
-            }
-            IngestMetadata ingestMetadata = metadata.custom(IngestMetadata.TYPE);
-            if (ingestMetadata == null) {
-                return 0;
-            }
-
-            int count = 0;
-            for (PipelineConfiguration configuration : ingestMetadata.getPipelines().values()) {
-                Map<String, Object> configMap = configuration.getConfigAsMap();
-                try {
-                    List<Map<String, Object>> processorConfigs = ConfigurationUtils.readList(null, null, configMap, PROCESSORS_KEY);
-                    for (Map<String, Object> processorConfigWithKey : processorConfigs) {
-                        for (Map.Entry<String, Object> entry : processorConfigWithKey.entrySet()) {
-                            count += numInferenceProcessors(entry.getKey(), entry.getValue());
-                        }
-                    }
-                    // We cannot throw any exception here. It might break other pipelines.
-                } catch (Exception ex) {
-                    logger.debug(() -> "failed gathering processors for pipeline [" + configuration.getId() + "]", ex);
-                }
+            try {
+                currentInferenceProcessors = InferenceProcessorInfoExtractor.countInferenceProcessors(state);
+            } catch (Exception ex) {
+                logger.debug("failed gathering processors for pipelines", ex);
```

Comment:
```suggestion
            } catch (Exception ex) {
                // We cannot throw any exception here. It might break other pipelines.
                logger.debug("failed gathering processors for pipelines", ex);
```